core/src/org/samtools/htsjdk/core/io/ByteSource.java

```diff
@@ -0,0 +1,107 @@
+package org.samtools.htsjdk.core.io;
+
+import java.nio.ByteBuffer;
+
+/**
+ * Interface for classes that produce binary data (i.e. bytes) that can be decoded
+ * into other datatypes.
+ *
+ * Methods should throw exceptions if:
+ * - The underlying data source/stream/channel is closed
```

Comment:
Agreed.  What are your thoughts on using something like https://github.com/Abnaxos/markdown-doclet so that we can write JavaDoc using MarkDown instead of HTML tags?

And yes, once we settle on the actual interface I 100% want to write highly detailed and well formatted javadoc here to try and set the bar for how our public classes should be documented.

Reply:
I like markdown, but most of the java folks that I know use HTML javadoc style; switching it to a custom doclet might produce some undocumented code or errors in rendering. Anyway, I am fine with both, as far as we enforce some style...